runtime/rococo/src/lib.rs

```diff
@@ -1766,7 +1766,7 @@ sp_api::impl_runtime_apis! {
 			proof: mmr::BatchProof<Hash>
 		) -> Result<(), mmr::Error> {
 			let nodes = leaves.into_iter().map(|leaf|mmr::DataOrHash::Data(leaf.into_opaque_leaf())).collect();
-			pallet_mmr::verify_leaves_proof::<MmrHashing, _>(root, nodes, proof)
+			pallet_mmr::verify_leaves_proof::<MmrHashing, _>(root, nodes, &proof)
```

Comment:
we own the data here, why pass by reference?

Reply:
A reference is only 8 bytes, while the entire structure is Sizeof(u64+u64+Vec) = 8 + 8 + 24. So there's less memory being copied, but it doesn't make much of a difference. As you pointed out, we would be copying all the hashes now, which would be more inefficient. Closing this PR for the moment.